packages/@ember/-internals/metal/lib/computed.ts

```diff
@@ -675,6 +676,25 @@ export class ComputedProperty extends ComputedDescriptor {
     let ret;
 
     try {
+      if (obj instanceof Ember.Component) {
+        // ensure two way binding works when the component has defined a computed
+        // property with both a setter and dependent keys, in that scenario without
+        // the sync observer added below the caller's value will never be updated
+        //
+        // See GH#18417 / GH#19028 for details.
+        let descriptor: ComputedDescriptor;
+        descriptor = descriptorForProperty(obj, keyName);
+        let meta = metaFor(obj);
+        if (
+          meta.isInitializing() &&
+          descriptor !== undefined &&
+          descriptor._dependentKeys !== undefined &&
+          descriptor._dependentKeys.length > 0
+        ) {
+          addObserver(obj, keyName, obj[PROPERTY_DID_CHANGE].bind(obj, keyName), undefined, true);
```

Comment:
I'd generally prefer to use an arrow function here (we don't have the same issue we had over in the component manager because this isn't in a loop) instead of `.bind`.

Reply:
We also have a bit of a logic issue here: we will add an observer for **every** set that happens.

Reply:
I thought that was the purpose of `meta.isInitializing`... we'll only do it on the initialization...